hie-plugin-api/Haskell/Ide/Engine/Compat.hs

```diff
@@ -27,3 +34,97 @@ isExtensionOf :: String -> FilePath -> Bool
 isExtensionOf ext@('.':_) = isSuffixOf ext . takeExtensions
 isExtensionOf ext         = isSuffixOf ('.':ext) . takeExtensions
 #endif
+
+
+#if MIN_VERSION_ghc(8, 4, 0)
+type GhcTc = GHC.GhcTc
+#else
+type GhcTc = GHC.Id
+#endif
+
+pattern HsOverLitType :: Type.Type -> GHC.HsExpr GhcTc
+pattern HsOverLitType t <-
+#if MIN_VERSION_ghc(8, 6, 0)
+    GHC.HsOverLit _ (GHC.overLitType -> t)
+#elif MIN_VERSION_ghc(8, 4, 0)
```

Comment:
Can just have the `else` clause here I think.

Reply:
Yeah, I wanted to ask that anyway, do we prefer easy to read, e.g. always three clauses although they often completely overlap, or conciseness when it comes to CPP?

Reply:
I prefer not having the duplication and just having two clauses when possible.